pets/peer-finder/peer-finder.go

```diff
@@ -40,7 +41,7 @@ var (
 	onStart   = flag.String("on-start", "", "Script to run on start, must accept a new line separated list of peers via stdin.")
 	svc       = flag.String("service", "", "Governing service responsible for the DNS records of the domain this pod is in.")
 	namespace = flag.String("ns", "", "The namespace this pod is running in. If unspecified, the POD_NAMESPACE env var is used.")
-	domain    = flag.String("domain", "cluster.local", "The Cluster Domain which is used by the Cluster.")
+	domain    = flag.String("domain", "", "The Cluster Domain which is used by the Cluster.")
```

Comment:
Can we keep the default as-is and just improve logging and add documentation to help debugging? Although the user must know of the domain option's existence, I think `cluster.local` works for a majority of clusters out there.